cmd/beemo/notify_reports.go

```diff
@@ -0,0 +1,130 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	comatproto "github.com/bluesky-social/indigo/api/atproto"
+	toolsozone "github.com/bluesky-social/indigo/api/ozone"
+	"github.com/bluesky-social/indigo/util"
+	"github.com/bluesky-social/indigo/xrpc"
+
+	"github.com/urfave/cli/v2"
+)
+
+func pollNewReports(cctx *cli.Context) error {
+	ctx := context.Background()
+	logger := configLogger(cctx, os.Stdout)
+	slackWebhookURL := cctx.String("slack-webhook-url")
+
+	// record last-seen report timestamp
+	since := time.Now()
+	// NOTE: uncomment this for testing
+	//since = time.Now().Add(time.Duration(-12) * time.Hour)
+	period := time.Duration(cctx.Int("poll-period")) * time.Second
+
+	// create a new session
+	xrpcc := &xrpc.Client{
+		Client: util.RobustHTTPClient(),
+		Host:   cctx.String("pds-host"),
+		Auth:   &xrpc.AuthInfo{Handle: cctx.String("handle")},
+	}
+
+	auth, err := comatproto.ServerCreateSession(ctx, xrpcc, &comatproto.ServerCreateSession_Input{
+		Identifier: xrpcc.Auth.Handle,
+		Password:   cctx.String("password"),
+	})
+	if err != nil {
+		return err
+	}
+	xrpcc.Auth.AccessJwt = auth.AccessJwt
+	xrpcc.Auth.RefreshJwt = auth.RefreshJwt
+	xrpcc.Auth.Did = auth.Did
+	xrpcc.Auth.Handle = auth.Handle
+
+	adminToken := cctx.String("admin-password")
+	if len(adminToken) > 0 {
+		xrpcc.AdminToken = &adminToken
+	}
+	logger.Info("report polling bot starting up...")
+	// can flip this bool to false to prevent spamming slack channel on startup
+	if true {
+		err := sendSlackMsg(ctx, fmt.Sprintf("restarted bot, monitoring for reports since `%s`...", since.Format(time.RFC3339)), slackWebhookURL)
+		if err != nil {
+			return err
+		}
+	}
+	for {
+		// refresh session
+		xrpcc.Auth.AccessJwt = xrpcc.Auth.RefreshJwt
+		refresh, err := comatproto.ServerRefreshSession(ctx, xrpcc)
+		if err != nil {
+			return err
+		}
+		xrpcc.Auth.AccessJwt = refresh.AccessJwt
+		xrpcc.Auth.RefreshJwt = refresh.RefreshJwt
+
+		// query just new reports (regardless of resolution state)
+		// ModerationQueryEvents(ctx context.Context, c *xrpc.Client, createdBy string, cursor string, includeAllUserRecords bool, limit int64, sortDirection string, subject string, types []string) (*ModerationQueryEvents_Output, error)
+		var limit int64 = 50
+		me, err := toolsozone.ModerationQueryEvents(
```

Comment:
lol lmao, love this method

Reply:
yeah, it's a bummer what is fairly reasonable/idiomatic for API design ends up so bad in codegen here. several ways we could re-architect the codegen some day. maybe worth creating a more ergonomic wrapper (with config struct) for a handful of these monster methods.